docs/flow/reference/components/discovery.relabel.md

```diff
@@ -6,15 +6,27 @@ title: discovery.relabel
 
 # discovery.relabel
 
+In Flow, targets are defined as sets of key-value pairs called _labels_.
+
 `discovery.relabel` rewrites the label set of the input targets by applying one
 or more `relabel_config` steps. If no relabeling steps are defined, then the
-input targets will be exported as-is.
+input targets are exported as-is.
 
 The most common use of `discovery.relabel` is to filter Prometheus targets or
-standardize the label set that will be passed to a downstream component. The
-`relabel_config` blocks will be applied to the label set of each target in
+standardize the label set that is passed to a downstream component. The
+`relabel_config` blocks are applied to the label set of each target in
 order of their appearance in the configuration file.
 
+Target labels which start with a double underscore `__` are considered as
+internal by Prometheus and are removed prior to scraping. To retain any of
+these labels post scraping, use a `labelmap` action. Service discovery
+mechanisms usually group their labels under `__meta_*`. For example, the
+discovery.kubernetes component populates a set of `__meta_kubernetes_*` labels
+to provide information about the discovered Kubernetes resources. If a
+relabeling step needs to store a label value temporarily, for example as the
+input to a subsequent relabeling step, use the `__tmp` label name prefix, as it
+is guaranteed to never be used by Prometheus itself.
```

Comment:
Can we remove the reference to Prometheus and scraping here to allow the discovery component to be logically decoupled from Prometheus? 

I think we can say something like:

> Target labels which start with a double underscore `__` are considered internal, and may be removed by other Flow components prior to telemetry collection. To retain any ...

Reply:
You're right. I've removed references to Prometheus and scraping, so that the description reads more naturally.

Do you think we should explain anything else related to relabeling here that a newcomer might find useful?

Reply:
Not that I can think of. This is a great first version to ship, and if anything comes up  we can add it later 👍